### ISSUE TITLE
router.base affecting proxy  behaviour in 2.11.1 and above

### ISSUE THREAD

**Issue by husayt** [nuxt/nuxt.js#7028]
### Version

[2.11.1-26376495.084a1ca](https://github.com/nuxt.js/releases/tag/2.11.1-26376495.084a1ca)

### Reproduction link

[https://github.com/nuxt/nuxt.js/pull/6999](https://github.com/nuxt/nuxt.js/pull/6999)

### Steps to reproduce

It affects proxy behavior when  `router.base` is defined.

The behavior changed somewhere between 2.11.0-26277131.7af9ec8c and
2.11.1-26376495.084a1ca

Before for the following configuration 

```js
 proxy: {
    "/api": {
      target: "http://myserver.com",
      pathRewrite: {
        "^/api": "/alpha"
      },
      changeOrigin: false,
      prependPath: false
    }
 },
 router: {
     base: "/alpha/"
 }
```

I would have `/api` pointing to `http://myserver.com/alpha` e.g. `/api/user` would  call `http://myserver.com/alpha/user`. But after the change  I can't access `/api`.

Without `router.base` defined it works as before, with `router.base` defined one only can define proxies starting with `base`

### What is expected ?

I would have `/api` pointing to `http://myserver.com/alpha` e.g. `/api/user` would  call `http://myserver.com/alpha/user`.

### What is actually happening?

After the change  I can't access `/api`.

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c10333">#c10333</a>)</em></sub></div>

**Comment by husayt** [comment]
This is effectively a breaking change affecting 2.12.0 release.

**Comment by pi0** [comment]
Hi, @husayt Thanks for the heads up. Regression fixed by 5a6cde3ebc23f04e89c30a4196a9b7d116b6d675. Proxy module was using an undocumented way to disable router.base (https://github.com/nuxt-community/proxy-module/blob/v1.3.3/lib/module.js#L64)

**Comment by husayt** [comment]
Excellent. Thanks
